vision_datasets/common/data_manifest.py

```diff
@@ -324,6 +342,62 @@ def sample_subset_by_ratio(self, sampling_ratio):
         sampled_images = [self.images[i] for i in sampled_image_ids]
         return DatasetManifest(sampled_images, self.labelmap, self.data_type)
 
+    def sample_n_way_n_shot(self, n_way, n_shot, random_seed=0):
+        """
+        Sample N-way N-shot dataset. Sampled dataset has n_way classes and each classes has n_shot images.
+        Each image will have only 1 class. If the original image has multiple classes, annotations for the other classes will be removed.
+
+        This method can throw a RuntimeError if it failed to generate N-way N-shot dataset.
+
+        Args:
+            n_way (int): The number of classes to sample.
+            n_shot (int): The number of images for each classes.
+
+        Returns:
+            A sampled dataset (DatasetManifest)
```

Comment:
Can we mention that an exception will be thrown if sampling failed?

Reply:
sure, will move the line 350 to here.